Electronic scale for Z axis probe

necdnk opened the pull request:
This sensor very sensitive. You can see it: http://www.youtube.com/watch?v=MtTQW5f7Cek
But sensitive can be changed. 
This solution is very precision (near 0,02mm on my 3d printer).
Description of implementation in Russian:
http://geektimes.ru/post/249138/

Algorithm of the calibration:
1. Set the sensor's sensitivity about 300 grams
2. Execute the first (hard) detection.
3. Set the sensor's sensitivity of about 5-50 grams.
4. Execute a second (accurate) detection.

necdnk committed:
Weight sensor as beds detector

necdnk committed:
Weight sensor as bed detector. Changes for auto bed leveling and z-min endstop

Commit:
HOMING_BUMP_DIVISOR Z axis = 40

Commit:
Clear code

thinkyhead committed:
Reverse the sign of zprobe_zoffset

- When editing on the LCD, minus now means below the nozzle
- Stores the opposite sign in EEPROM, so bumped to “V20”

AnHardt committed:
Respect DISABLE_AXIS settings when DEFAULT_STEPPER_DEACTIVE_TIME is over

instead of simply turn off all.

fmalpartida committed:
Updated SAV MkI pin header.

Wackerbarth committed:
Deprecate Support for Arduino IDE < 1.0.5


Note that newer versions of the IDE can still
be made to support compilation for the older boards.

Wackerbarth committed:
Deprecate Support for Arduino IDE < 1.0.5


Note that newer versions of the IDE can still
be made to support compilation for the older boards.

boelle committed:
Merge pull request #2257 from fmalpartida/bugFixing

Updated SAV MkI extension module header connector.

thinkyhead committed:
Include "Z" in M851 report

thinkyhead committed:
Merge pull request #2167 from thinkyhead/reverse_zoffset_sign

Reverse the sign of zprobe_zoffset

thinkyhead committed:
Parse N[0-9]+[ ]* differently

More general solution to skip N[0-9]+[ ]* in the parser as in #2218

thinkyhead committed:
Merge pull request #2263 from thinkyhead/skip_n_digits

Parse N[0-9]+[ ]* differently

Wackerbarth committed:
Merge pull request #2260 from Wackerbarth/DropArduino0.x.x

Deprecate Support for Arduino IDE < 1.0.5

Commit:
Fix volumetric multiplier display for filament sensor

The volumetric multiplier display needs to be multiplied by 100 in order to make it an integer for display purposes.

AnHardt committed:
Optimise out_of_range_error()

by avoiding the strncopy()

thinkyhead committed:
Prevent current_command_args from bypassing nul

thinkyhead committed:
Merge pull request #2270 from thinkyhead/skip_n_digits

Properly handle M117 with no message

thinkyhead committed:
Set auto fan pins to OUTPUT during init

Fixes the problem of these pins being left in INPUT state, as described
in #1497

thinkyhead committed:
Formatting tweaks as part of fixing issue 1497

schlotzz committed:
Enable ZigZag AutoBedLeveling on Deltas

ZigZag has been disabled by fault due to do_topography_map

schlotzz committed:
AllenKey probe lifted twice by Z_RAISE_BETWEEN_PROBINGS

Z_RAISE_BETWEEN_PROBINGS is being applied twice by fault. Removed
unnecessary block in run_z_probe()

MagoKimbra committed:
Fix error next command in autotune

Now send only temperature for T or B, but not send OK.
Host interprets the line to show the right temperature, but not in
response to M105 then stop and send commands until it ends the autotune.

tonnico committed:
make the dual z-endstop optional
it's disabled by default

tonnico committed:
Merge pull request #2277 from Wurstnase/z2_endstop_optional

make the dual z-endstop optional

AnHardt committed:
Merge pull request #2216 from AnHardt/disableSteppers

Respect DISABLE_AXIS settings when DEFAULT_STEPPER_DEACTIVE_TIME is over

thinkyhead committed:
Use translated strings for axis movement

Suggest we use the translated strings here. They use (except for
German) the phrasing “Move X” instead of just the axis letter, but they
should fit the available space. The “Extruder” string would be the
same, except it will be translated.

thinkyhead committed:
Merge pull request #2264 from filipmu/patch-1

Fix volumetric multiplier display for filament sensor

thinkyhead committed:
Don't re-apply Z_RAISE_BETWEEN_PROBINGS in probe_pt

thinkyhead committed:
Merge pull request #2265 from AnHardt/PositionOOR

Optimize out_of_range_error() ftw!

thinkyhead committed:
Add SanityCheck for bed probe position settings

thinkyhead committed:
Consistent punctuation in SanityCheck errors

thinkyhead committed:
Move Z Safe Homing point to the middle of the movement range

thinkyhead committed:
Fix configurator section directives and values lists

thinkyhead committed:
Fix up spacing for filament sensor values

thinkyhead committed:
Clean up spacing and comments

Also clean up some trailing spaces in a few other sources

thinkyhead committed:
Cleanup spaces and comments in configs and conditionals

thinkyhead committed:
Move extruder offset settings next to extruders setting

thinkyhead committed:
Move Delta settings lower in delta files

thinkyhead committed:
Fix up allen key settings in delta configs

thinkyhead committed:
Update delta configs with latest settings

thinkyhead committed:
Merge pull request #2274 from MagoKimbra/autotune_error_next_command

Fix error next command in autotune

Wackerbarth committed:
Merge pull request #2283 from thinkyhead/sanitycheck_punctuation

Consistent punctuation in SanityCheck errors

thinkyhead committed:
Merge pull request #2281 from thinkyhead/fix_double_raise_between

Don't re-apply Z_RAISE_BETWEEN_PROBINGS in probe_pt

Wackerbarth committed:
Merge pull request #2280 from thinkyhead/ultralcd_language

Use translated strings for axis movement

AnHardt committed:
Rework buzzing

by:
Moving HAS_LCD_BUZZ macro to Coditionals.h
Renaming HAS_LCD_BUZZ to HAS_BUZZER to make clear is has nothing to do with the lcd.
Removing the ULTRALCD condition.

Moving declaration of lcd_buzz() out of the ULTRA_LCD block in ultralcd.h
Moving definition of lcd_buzz() out of the ULTIPANEL block in ultralcd.cpp
Renaming lcd_buzz() to buzz() to make clear is has nothing to do with the lcd.

All buzzing code is now only dependent on the existence of a BEEPER-pin or the definition of a LCD_USE_I2C_BUZZER.

To do: Check the conditions for the BEEPER-pin in all pin-files.

AnHardt committed:
Merge pull request #2271 from thinkyhead/fix_issue_1497

Set auto fan pins to OUTPUT during init

thinkyhead committed:
Patch up more outdated config lines

thinkyhead committed:
Fix ULTRA_LCD spelling in HAS_LCD_BUZZ

thinkyhead committed:
Remove a test from pins_RAMPS_13.h

- Add 3 new lines to `pins.h` to remove 2 redundant tests in
`pins_RAMPS_13.h`

thinkyhead committed:
Merge pull request #2296 from AnHardt/iss2289-4

Rework buzzing V2

Commit:
Merge branch 'Development' of https://github.com/necdnk/Marlin into Development

AnHardt committed:
Add lost MKS_BASE  to pins.h

Wackerbarth committed:
Merge pull request #2297 from AnHardt/MKS

Add lost MKS_BASE  to pins.h

thinkyhead committed:
Add pins_RAMPS_13_EFB.h to reduce redundancy

thinkyhead committed:
Merge pull request #2295 from thinkyhead/less_efb

Remove a test from pins_RAMPS_13.h

thinkyhead committed:
Merge pull request #2285 from thinkyhead/fixup_configs

Bring configurations up-to-date

thinkyhead committed:
Merge pull request #2282 from thinkyhead/sanitycheck_probe_pos

Add SanityCheck for bed probe position settings

thinkyhead committed:
Merge pull request #2273 from schlotzz/Development

Enable ZigZag AutoBedLeveling on Deltas

thinkyhead committed:
Check the temperature before filament change

thinkyhead committed:
Replace target with destination in M600

thinkyhead committed:
Cleanup M600 documentation comments

thinkyhead committed:
Use millis for M600 audio feedback interval

Commit:
!!!Not tested on the device!!!
changes after codereview

AnHardt committed:
Rename and repair MELZI_1284

to MELZI_MAKR3D.

Fix for #2288
As the normal MELZI also can have a 1284.

necdnk committed:
Tested on the my printer. bugfix

AnHardt committed:
Move buzzing code to buzzr.h & buzzer.cpp

at least the lcd independent part from Marlin_main.cpp.

necdnk committed:
tested! after review
it is working

oysteinkrog commented:
How well does it handle plastic on the nozzle?
Does it work with bed surfaces that are not conductive? (e.g. covered in abs juice).

necdnk commented:
You can use any surface (glass, metal, Kapton, ABS juice)
In this video, i use mirror:
http://www.youtube.com/watch?v=b3wcx6VSzj4
If you are using ABS juice, you can put in the slicer z-offset height value of this layer.
Starting the calibration must be done on the heated extruder. In this case, you get a very precise adjustment of the height.
Ask questions, I'm happy to answer them.
Sorry for my English ;)

tonnico commented on Marlin/Configuration_adv.h:
```diff
@@ -192,7 +192,7 @@
 #define Y_HOME_BUMP_MM 5
 #define Z_HOME_BUMP_MM 2
-#define HOMING_BUMP_DIVISOR {2, 2, 4}  // Re-Bump Speed Divisor (Divides the Homing Feedrate)
+#define HOMING_BUMP_DIVISOR {2, 2, 40}  // Re-Bump Speed Divisor (Divides the Homing Feedrate)
```
don't change this in the standard configs

tonnico commented on Marlin/Marlin_main.cpp:
```diff
@@ -54,6 +54,7 @@
 #include "language.h"
 #include "pins_arduino.h"
 #include "math.h"
+#include "HX711.h"
```
#ifdef ELECTRONIC_SCALE_PROBE?

necdnk committed:
some changes for pull request

tonnico commented:
You must update your fork before merge. The endstop logic changed some weeks ago. There are no booleans anymore for the endstops.

thinkyhead committed:
Merge pull request #2303 from AnHardt/MAKR3D

Rename and repair MELZI_1284

thinkyhead committed:
Merge pull request #2307 from AnHardt/buzz20

Move buzzing code to buzzr.h & buzzer.cpp

thinkyhead commented:
@necdnk You'll step all over your own toes making changes directly to your Development branch. You should always make a new branch for PRs. For example, `git checkout Development -b my_feature_branch`. But before you do that you'll need to get the freshest Development branch. For example, `git fetch upstream ; git checkout Development ; git merge upstream/Development`.

thinkyhead committed:
Updated Russian language

Gleaned from
https://github.com/MarlinFirmware/Marlin/commit/4ac14a7e5fa8e6e395436664
8976eafd1ae220ae

Wackerbarth committed:
Merge pull request #2325 from thinkyhead/russian_lang

Updated Russian language

oysteinkrog commented:
I think this is a pretty nice way to do distance measurement, I especially like that the measurement is done with the actual nozzle itself (without applying too much force etc).
I think I will order the parts and try is myself.
@necdnk Is there a reason you used the 5kg load cell instead of the 3kg?
I would think 3kg (or maybe the 1kg model too) should be enough?
http://www.aliexpress.com/item/Kitchen-Digital-Electronic-Scale-Pressure-Sensor-Load-Cell-YZC-131-3Kg-3-Hole/32320250538.html

necdnk commented:
@oysteinkrog yes! you can try to use 3kg load cell. I think this load cells (3kg and 5kg) is equals. 
You can try using 1kg sensor, but it is advisable not to exceed the pressure in 2kg.
I pulled it out of, from destroyed kitchen scales. But later I bought several sensors from China, and this is also working successfully.
I want to try the tensoresistors used in these sensors like this: http://ru.aliexpress.com/item/Free-shipping-5pcs-lot-BF350-3AA-BF350-Precision-resistive-strain-gauge-strain-gauge-for-the-pressure/1935464167.html
They are very small, but need to learn how to use them correctly.

necdnk commented:
@thinkyhead I will need to make a new PR? From the new branch? And this PR to be closed?
I apologize for such questions. This is my first PR, before I did not work with them.

Wackerbarth commented:
@necdnk -- No need for a new PR. Just rebase the code to the current head of the MarlinFirmware Development branch, make any revisions needed, and push it to your github account. Automatically, this PR will be updated by the github infrastructure.

For your future reference, when starting a new feature or patch, start a new branch within your repository, one per issue. That way you can easily work on multiple issues at the same time. When the issue is closed, you should delete the branch so that you don't clutter up your branch namespace with out-of-date, now irrelevant, labels.

nallath committed:
Added correct dutch translation for calibrate.

nallath committed:
Update Ultimaker URL

Wackerbarth committed:
Merge pull request #2331 from Wackerbarth/Ultimaker

Ultimaker

thinkyhead commented:
My preferred method to fix up your type of situation would be:

```
git fetch upstream
git checkout Development
git merge upstream/Development ;# merges are usually easier
git checkout upstream/Development -b my_clean_feature
git diff my_clean_feature Development | git apply
```

Then you can cherry-pick the feature branch to create separate commits, or just make one large commit.

At this point your Development branch will still be out of sync with upstream, so you will want to reset it to match the upstream Development HEAD.

thinkyhead committed:
Merge pull request #2301 from thinkyhead/improve_M600

Improve M600 slightly

Wackerbarth commented:
"or just make one large commit." -- Actually, breaking your commit into smaller pieces has advantages.
(1) For someone reviewing the code, smaller pieces are easier to understand.
(2) Conflict resolution is also easier to manage.

thinkyhead committed:
Init all endstop bits to 0

AnHardt committed:
Merge pull request #2332 from thinkyhead/init_endstop_bits

Init endstop bits to 0

Commit:
Electronic_scale_Z_probe

Commit:
Merge branch 'Development' of https://github.com/necdnk/Marlin into Electronic_scale_Z_probe

Conflicts:
	Marlin/Configuration.h
	Marlin/Marlin_main.cpp
	Marlin/pins_RAMPS_13.h
	Marlin/stepper.cpp

Commit:
Remove initialization of members

Commit:
Remove unnecessary changes

thinkyhead commented:
> breaking your commit into smaller pieces has advantages

Truly. I particularly like how cherry-picking allows you to group changes that should go together, which otherwise the diff algorithm might not be able to figure out.

thinkyhead commented on Marlin/Configuration.h:
```diff
@@ -544,6 +544,14 @@ const bool Z_PROBE_ENDSTOP_INVERTING = false; // set to true to invert the logic
+	#define SCALE_GAIN 64 // 128 or 64 only for A channel 32 only for B channel
+	#define SCALE_SENSIVITY_HARD 40
+	#define SCALE_SENSIVITY_ACCURATE 5
+	#define SCALE_DIVISOR 1000.f
```
Two spaces, no tabs.

http://www.marlinfirmware.org/index.php/Coding_Standards

thinkyhead commented on Marlin/HX711.h:
```diff
@@ -0,0 +1,88 @@
+#define __HX711_h
+
+#if ARDUINO >= 100
+#include "Arduino.h"
```
Indent items between `#if ... #endif`.

http://www.marlinfirmware.org/index.php/Coding_Standards

boelle commented:
Please rebase, the code cant be merged as is

thinkyhead committed:
Merge pull request #2269 from Wackerbarth/SwitchToArduinoIDE1.6.4

Switch to Arduino IDE 1.6.4  for Travis testing

thinkyhead committed:
Cleanup of comments & spacing

thinkyhead committed:
Fix comment on M83, add comment to prepare_move

Wackerbarth committed:
Clean up comments and spacing

Commit:
It has not been tested on the hardware

Commit:
Merge remote-tracking branch 'remotes/upstream/Development' into Electronic_scale_Z_probe

Conflicts:
	.travis.yml
	ArduinoAddons/Arduino_1.6.x/hardware/marlin/avr/cores/arduino/Server.h
	ArduinoAddons/Arduino_1.6.x/hardware/marlin/avr/cores/arduino/Tone.cpp
	ArduinoAddons/Arduino_1.6.x/hardware/marlin/avr/cores/arduino/WCharacter.h
	ArduinoAddons/Arduino_1.6.x/hardware/marlin/avr/cores/arduino/WInterrupts.c
	ArduinoAddons/Arduino_1.6.x/hardware/marlin/avr/cores/arduino/WMath.cpp
	ArduinoAddons/Arduino_1.6.x/hardware/marlin/avr/cores/arduino/binary.h
	ArduinoAddons/Arduino_1.6.x/hardware/marlin/avr/cores/arduino/new.h
	ArduinoAddons/Arduino_1.6.x/hardware/marlin/avr/cores/arduino/wiring.c
	ArduinoAddons/Arduino_1.6.x/hardware/marlin/avr/cores/arduino/wiring_analog.c
	ArduinoAddons/Arduino_1.6.x/hardware/marlin/avr/cores/arduino/wiring_digital.c
	ArduinoAddons/Arduino_1.6.x/hardware/marlin/avr/cores/arduino/wiring_private.h
	ArduinoAddons/Arduino_1.6.x/hardware/marlin/avr/cores/arduino/wiring_pulse.c
	ArduinoAddons/Arduino_1.6.x/hardware/marlin/avr/cores/arduino/wiring_shift.c
	ArduinoAddons/Arduino_1.6.x/hardware/marlin/avr/cores/at90usb/HardwareSerial.h
	ArduinoAddons/Arduino_1.6.x/hardware/marlin/avr/cores/at90usb/Tone.cpp
	ArduinoAddons/Arduino_1.6.x/hardware/marlin/avr/cores/at90usb/WCharacter.h
	ArduinoAddons/Arduino_1.6.x/hardware/marlin/avr/cores/at90usb/WInterrupts.c
	ArduinoAddons/Arduino_1.6.x/hardware/marlin/avr/cores/at90usb/WMath.cpp
	ArduinoAddons/Arduino_1.6.x/hardware/marlin/avr/cores/at90usb/binary.h
	ArduinoAddons/Arduino_1.6.x/hardware/marlin/avr/cores/at90usb/wiring.c
	ArduinoAddons/Arduino_1.6.x/hardware/marlin/avr/cores/at90usb/wiring_analog.c
	ArduinoAddons/Arduino_1.6.x/hardware/marlin/avr/cores/at90usb/wiring_digital.c
	ArduinoAddons/Arduino_1.6.x/hardware/marlin/avr/cores/at90usb/wiring_private.h
	ArduinoAddons/Arduino_1.6.x/hardware/marlin/avr/cores/at90usb/wiring_pulse.c
	ArduinoAddons/Arduino_1.6.x/hardware/marlin/avr/cores/at90usb/wiring_shift.c
	ArduinoAddons/Arduino_1.6.x/hardware/marlin/avr/libraries/LiquidCrystal/keywords.txt
	ArduinoAddons/Arduino_1.6.x/hardware/marlin/avr/libraries/LiquidCrystal/src/LiquidCrystal.cpp
	ArduinoAddons/Arduino_1.6.x/hardware/marlin/avr/libraries/LiquidCrystal/src/LiquidCrystal.h
	ArduinoAddons/Arduino_1.6.x/hardware/marlin/avr/libraries/SPI/SPI.cpp
	ArduinoAddons/Arduino_1.6.x/hardware/marlin/avr/libraries/SPI/SPI.h
	ArduinoAddons/Arduino_1.6.x/hardware/marlin/avr/libraries/SPI/keywords.txt
	ArduinoAddons/Arduino_1.6.x/hardware/marlin/avr/libraries/U8glib/ChangeLog
	ArduinoAddons/Arduino_1.6.x/hardware/marlin/avr/libraries/U8glib/INSTALL.TXT
	ArduinoAddons/Arduino_1.6.x/hardware/marlin/avr/libraries/U8glib/U8glib.cpp
	ArduinoAddons/Arduino_1.6.x/hardware/marlin/avr/libraries/U8glib/U8glib.h
	ArduinoAddons/Arduino_1.6.x/hardware/marlin/avr/libraries/U8glib/examples/Bitmap/Bitmap.ino
	ArduinoAddons/Arduino_1.6.x/hardware/marlin/avr/libraries/U8glib/examples/Chess/Chess.ino
	ArduinoAddons/Arduino_1.6.x/hardware/marlin/avr/libraries/U8glib/examples/Console/Console.ino
	ArduinoAddons/Arduino_1.6.x/hardware/marlin/avr/libraries/U8glib/examples/F/F.ino
	ArduinoAddons/Arduino_1.6.x/hardware/marlin/avr/libraries/U8glib/examples/GraphicsTest/GraphicsTest.ino
	ArduinoAddons/Arduino_1.6.x/hardware/marlin/avr/libraries/U8glib/examples/HelloWorld/HelloWorld.ino
	ArduinoAddons/Arduino_1.6.x/hardware/marlin/avr/libraries/U8glib/examples/Menu/Menu.ino
	ArduinoAddons/Arduino_1.6.x/hardware/marlin/avr/libraries/U8glib/examples/PrintTest/PrintTest.ino
	ArduinoAddons/Arduino_1.6.x/hardware/marlin/avr/libraries/U8glib/examples/Rotation/Rotation.ino
	ArduinoAddons/Arduino_1.6.x/hardware/marlin/avr/libraries/U8glib/examples/Scale/Scale.ino
	ArduinoAddons/Arduino_1.6.x/hardware/marlin/avr/libraries/U8glib/examples/TextRotX/TextRotX.ino
	ArduinoAddons/Arduino_1.6.x/hardware/marlin/avr/libraries/U8glib/examples/U8gLogo/U8gLogo.ino
	ArduinoAddons/Arduino_1.6.x/hardware/marlin/avr/libraries/U8glib/examples/XBM/XBM.ino
	ArduinoAddons/Arduino_1.6.x/hardware/marlin/avr/libraries/U8glib/license.txt
	ArduinoAddons/Arduino_1.6.x/hardware/marlin/avr/libraries/U8glib/utility/chessengine.c
	ArduinoAddons/Arduino_1.6.x/hardware/marlin/avr/libraries/U8glib/utility/u8g_bitmap.c
	ArduinoAddons/Arduino_1.6.x/hardware/marlin/avr/libraries/U8glib/utility/u8g_circle.c
	ArduinoAddons/Arduino_1.6.x/hardware/marlin/avr/libraries/U8glib/utility/u8g_clip.c
	ArduinoAddons/Arduino_1.6.x/hardware/marlin/avr/libraries/U8glib/utility/u8g_com_api.c
	ArduinoAddons/Arduino_1.6.x/hardware/marlin/avr/libraries/U8glib/utility/u8g_com_api_16gr.c
	ArduinoAddons/Arduino_1.6.x/hardware/marlin/avr/libraries/U8glib/utility/u8g_com_arduino_common.c
	ArduinoAddons/Arduino_1.6.x/hardware/marlin/avr/libraries/U8glib/utility/u8g_com_arduino_fast_parallel.c
	ArduinoAddons/Arduino_1.6.x/hardware/marlin/avr/libraries/U8glib/utility/u8g_com_arduino_hw_usart_spi.c
	ArduinoAddons/Arduino_1.6.x/hardware/marlin/avr/libraries/U8glib/utility/u8g_com_arduino_no_en_parallel.c
	ArduinoAddons/Arduino_1.6.x/hardware/marlin/avr/libraries/U8glib/utility/u8g_com_arduino_parallel.c
	ArduinoAddons/Arduino_1.6.x/hardware/marlin/avr/libraries/U8glib/utility/u8g_com_arduino_port_d_wr.c
	ArduinoAddons/Arduino_1.6.x/hardware/marlin/avr/libraries/U8glib/utility/u8g_com_arduino_ssd_i2c.c
	ArduinoAddons/Arduino_1.6.x/hardware/marlin/avr/libraries/U8glib/utility/u8g_com_arduino_st7920_hw_spi.c
	ArduinoAddons/Arduino_1.6.x/hardware/marlin/avr/libraries/U8glib/utility/u8g_com_arduino_st7920_spi.c
	ArduinoAddons/Arduino_1.6.x/hardware/marlin/avr/libraries/U8glib/utility/u8g_com_arduino_std_sw_spi.c
	ArduinoAddons/Arduino_1.6.x/hardware/marlin/avr/libraries/U8glib/utility/u8g_com_arduino_sw_spi.c
	ArduinoAddons/Arduino_1.6.x/hardware/marlin/avr/libraries/U8glib/utility/u8g_com_arduino_t6963.c
	ArduinoAddons/Arduino_1.6.x/hardware/marlin/avr/libraries/U8glib/utility/u8g_com_atmega_hw_spi.c
	ArduinoAddons/Arduino_1.6.x/hardware/marlin/avr/libraries/U8glib/utility/u8g_com_atmega_parallel.c
	ArduinoAddons/Arduino_1.6.x/hardware/marlin/avr/libraries/U8glib/utility/u8g_com_atmega_st7920_hw_spi.c
	ArduinoAddons/Arduino_1.6.x/hardware/marlin/avr/libraries/U8glib/utility/u8g_com_atmega_st7920_spi.c
	ArduinoAddons/Arduino_1.6.x/hardware/marlin/avr/libraries/U8glib/utility/u8g_com_atmega_sw_spi.c
	ArduinoAddons/Arduino_1.6.x/hardware/marlin/avr/libraries/U8glib/utility/u8g_com_io.c
	ArduinoAddons/Arduino_1.6.x/hardware/marlin/avr/libraries/U8glib/utility/u8g_com_null.c
	ArduinoAddons/Arduino_1.6.x/hardware/marlin/avr/libraries/U8glib/utility/u8g_cursor.c
	ArduinoAddons/Arduino_1.6.x/hardware/marlin/avr/libraries/U8glib/utility/u8g_delay.c
	ArduinoAddons/Arduino_1.6.x/hardware/marlin/avr/libraries/U8glib/utility/u8g_dev_flipdisc_2x7.c
	ArduinoAddons/Arduino_1.6.x/hardware/marlin/avr/libraries/U8glib/utility/u8g_dev_gprof.c
	ArduinoAddons/Arduino_1.6.x/hardware/marlin/avr/libraries/U8glib/utility/u8g_dev_ili9325d_320x240.c
	ArduinoAddons/Arduino_1.6.x/hardware/marlin/avr/libraries/U8glib/utility/u8g_dev_ks0108_128x64.c
	ArduinoAddons/Arduino_1.6.x/hardware/marlin/avr/libraries/U8glib/utility/u8g_dev_lc7981_160x80.c
	ArduinoAddons/Arduino_1.6.x/hardware/marlin/avr/libraries/U8glib/utility/u8g_dev_lc7981_240x128.c
	ArduinoAddons/Arduino_1.6.x/hardware/marlin/avr/libraries/U8glib/utility/u8g_dev_lc7981_240x64.c
	ArduinoAddons/Arduino_1.6.x/hardware/marlin/avr/libraries/U8glib/utility/u8g_dev_lc7981_320x64.c
	ArduinoAddons/Arduino_1.6.x/hardware/marlin/avr/libraries/U8glib/utility/u8g_dev_null.c
	ArduinoAddons/Arduino_1.6.x/hardware/marlin/avr/libraries/U8glib/utility/u8g_dev_pcd8544_84x48.c
	ArduinoAddons/Arduino_1.6.x/hardware/marlin/avr/libraries/U8glib/utility/u8g_dev_pcf8812_96x65.c
	ArduinoAddons/Arduino_1.6.x/hardware/marlin/avr/libraries/U8glib/utility/u8g_dev_sbn1661_122x32.c
	ArduinoAddons/Arduino_1.6.x/hardware/marlin/avr/libraries/U8glib/utility/u8g_dev_ssd1306_128x32.c
	ArduinoAddons/Arduino_1.6.x/hardware/marlin/avr/libraries/U8glib/utility/u8g_dev_ssd1306_128x64.c
	ArduinoAddons/Arduino_1.6.x/hardware/marlin/avr/libraries/U8glib/utility/u8g_dev_ssd1309_128x64.c
	ArduinoAddons/Arduino_1.6.x/hardware/marlin/avr/libraries/U8glib/utility/u8g_dev_ssd1322_nhd31oled_bw.c
	ArduinoAddons/Arduino_1.6.x/hardware/marlin/avr/libraries/U8glib/utility/u8g_dev_ssd1322_nhd31oled_gr.c
	ArduinoAddons/Arduino_1.6.x/hardware/marlin/avr/libraries/U8glib/utility/u8g_dev_ssd1325_nhd27oled_bw.c
	ArduinoAddons/Arduino_1.6.x/hardware/marlin/avr/libraries/U8glib/utility/u8g_dev_ssd1325_nhd27oled_bw_new.c
	ArduinoAddons/Arduino_1.6.x/hardware/marlin/avr/libraries/U8glib/utility/u8g_dev_ssd1325_nhd27oled_gr.c
	ArduinoAddons/Arduino_1.6.x/hardware/marlin/avr/libraries/U8glib/utility/u8g_dev_ssd1325_nhd27oled_gr_new.c
	ArduinoAddons/Arduino_1.6.x/hardware/marlin/avr/libraries/U8glib/utility/u8g_dev_ssd1327_96x96_gr.c
	ArduinoAddons/Arduino_1.6.x/hardware/marlin/avr/libraries/U8glib/utility/u8g_dev_st7565_64128n.c
	ArduinoAddons/Arduino_1.6.x/hardware/marlin/avr/libraries/U8glib/utility/u8g_dev_st7565_dogm128.c
	ArduinoAddons/Arduino_1.6.x/hardware/marlin/avr/libraries/U8glib/utility/u8g_dev_st7565_dogm132.c
	ArduinoAddons/Arduino_1.6.x/hardware/marlin/avr/libraries/U8glib/utility/u8g_dev_st7565_lm6059.c
	ArduinoAddons/Arduino_1.6.x/hardware/marlin/avr/libraries/U8glib/utility/u8g_dev_st7565_lm6063.c
	ArduinoAddons/Arduino_1.6.x/hardware/marlin/avr/libraries/U8glib/utility/u8g_dev_st7565_nhd_c12832.c
	ArduinoAddons/Arduino_1.6.x/hardware/marlin/avr/libraries/U8glib/utility/u8g_dev_st7565_nhd_c12864.c
	ArduinoAddons/Arduino_1.6.x/hardware/marlin/avr/libraries/U8glib/utility/u8g_dev_st7687_c144mvgd.c
	ArduinoAddons/Arduino_1.6.x/hardware/marlin/avr/libraries/U8glib/utility/u8g_dev_st7920_128x64.c
	ArduinoAddons/Arduino_1.6.x/hardware/marlin/avr/libraries/U8glib/utility/u8g_dev_st7920_192x32.c
	ArduinoAddons/Arduino_1.6.x/hardware/marlin/avr/libraries/U8glib/utility/u8g_dev_st7920_202x32.c
	ArduinoAddons/Arduino_1.6.x/hardware/marlin/avr/libraries/U8glib/utility/u8g_dev_t6963_128x64.c
	ArduinoAddons/Arduino_1.6.x/hardware/marlin/avr/libraries/U8glib/utility/u8g_dev_t6963_240x128.c
	ArduinoAddons/Arduino_1.6.x/hardware/marlin/avr/libraries/U8glib/utility/u8g_dev_t6963_240x64.c
	ArduinoAddons/Arduino_1.6.x/hardware/marlin/avr/libraries/U8glib/utility/u8g_dev_tls8204_84x48.c
	ArduinoAddons/Arduino_1.6.x/hardware/marlin/avr/libraries/U8glib/utility/u8g_dev_uc1610_dogxl160.c
	ArduinoAddons/Arduino_1.6.x/hardware/marlin/avr/libraries/U8glib/utility/u8g_dev_uc1701_dogs102.c
	ArduinoAddons/Arduino_1.6.x/hardware/marlin/avr/libraries/U8glib/utility/u8g_dev_uc1701_mini12864.c
	ArduinoAddons/Arduino_1.6.x/hardware/marlin/avr/libraries/U8glib/utility/u8g_font.c
	ArduinoAddons/Arduino_1.6.x/hardware/marlin/avr/libraries/U8glib/utility/u8g_font_data.c
	ArduinoAddons/Arduino_1.6.x/hardware/marlin/avr/libraries/U8glib/utility/u8g_line.c
	ArduinoAddons/Arduino_1.6.x/hardware/marlin/avr/libraries/U8glib/utility/u8g_ll_api.c
	ArduinoAddons/Arduino_1.6.x/hardware/marlin/avr/libraries/U8glib/utility/u8g_page.c
	ArduinoAddons/Arduino_1.6.x/hardware/marlin/avr/libraries/U8glib/utility/u8g_pb.c
	ArduinoAddons/Arduino_1.6.x/hardware/marlin/avr/libraries/U8glib/utility/u8g_pb14v1.c
	ArduinoAddons/Arduino_1.6.x/hardware/marlin/avr/libraries/U8glib/utility/u8g_pb16h1.c
	ArduinoAddons/Arduino_1.6.x/hardware/marlin/avr/libraries/U8glib/utility/u8g_pb16h2.c
	ArduinoAddons/Arduino_1.6.x/hardware/marlin/avr/libraries/U8glib/utility/u8g_pb16v1.c
	ArduinoAddons/Arduino_1.6.x/hardware/marlin/avr/libraries/U8glib/utility/u8g_pb16v2.c
	ArduinoAddons/Arduino_1.6.x/hardware/marlin/avr/libraries/U8glib/utility/u8g_pb32h1.c
	ArduinoAddons/Arduino_1.6.x/hardware/marlin/avr/libraries/U8glib/utility/u8g_pb8h1.c
	ArduinoAddons/Arduino_1.6.x/hardware/marlin/avr/libraries/U8glib/utility/u8g_pb8h1f.c
	ArduinoAddons/Arduino_1.6.x/hardware/marlin/avr/libraries/U8glib/utility/u8g_pb8h2.c
	ArduinoAddons/Arduino_1.6.x/hardware/marlin/avr/libraries/U8glib/utility/u8g_pb8h8.c
	ArduinoAddons/Arduino_1.6.x/hardware/marlin/avr/libraries/U8glib/utility/u8g_pb8v1.c
	ArduinoAddons/Arduino_1.6.x/hardware/marlin/avr/libraries/U8glib/utility/u8g_pb8v2.c
	ArduinoAddons/Arduino_1.6.x/hardware/marlin/avr/libraries/U8glib/utility/u8g_rect.c
	ArduinoAddons/Arduino_1.6.x/hardware/marlin/avr/libraries/U8glib/utility/u8g_rot.c
	ArduinoAddons/Arduino_1.6.x/hardware/marlin/avr/libraries/U8glib/utility/u8g_scale.c
	ArduinoAddons/Arduino_1.6.x/hardware/marlin/avr/libraries/U8glib/utility/u8g_state.c
	ArduinoAddons/Arduino_1.6.x/hardware/marlin/avr/libraries/U8glib/utility/u8g_u16toa.c
	ArduinoAddons/Arduino_1.6.x/hardware/marlin/avr/libraries/U8glib/utility/u8g_u8toa.c
	ArduinoAddons/Arduino_1.6.x/hardware/marlin/avr/libraries/U8glib/utility/u8g_virtual_screen.c
	ArduinoAddons/Arduino_1.6.x/hardware/marlin/avr/variants/mega/pins_arduino.h
	Marlin/Marlin_main.cpp
	Marlin/SanityCheck.h
	Marlin/buzzer.h

necdnk committed:
Repair. Tested on the printer.

necdnk committed:
Merge branch 'Electronic_scale_Z_probe' into Development

Conflicts:
	Marlin/stepper.cpp

necdnk commented:
This code tested on my printer and ready for merge.
Please check it.

thinkyhead commented:
The changes to these files seem strange, in part because the diffs are flagging lines that seem to be the same:

```
ArduinoAddons/Arduino_1.6.x/hardware/marlin/avr/libraries/U8glib/utility/u8g_circle.c
ArduinoAddons/Arduino_1.6.x/hardware/marlin/avr/libraries/U8glib/utility/u8g_dev_ssd1309_128x64.c
```

necdnk commented:
@thinkyhead I think it's a mistake comparing git :(
I merge changes from upstream/Development into my repository.

oysteinkrog commented:
You need to rebase this, it's a mess right now.

thinkyhead commented:
@necdnk Are you able to redo this? You'll need to create a separate branch (not named "Development") and rebuild your feature there.

So first make a copy of your branch with its commits:

```
git checkout Development -b my_feature_branch
```

Then you might do:

```
git remote add upstream https://github.com/MarlinFirmware/Marlin.git ;# if needed
git fetch upstream
git checkout Development
git reset --hard upstream/Development
```

And finally:

```
git checkout my_feature_branch
git rebase Development
```

To fix up your fork's `Development` branch as stored on Github you will need to change your fork's default branch to something other than `Development` using the [Settings link for your fork](https://github.com/necdnk/Marlin/settings), then delete the `Development` branch, and then finally push your local `Development` branch to `origin/Development`. Something along the lines of:

```
git checkout Development
git branch --unset-upstream
git push origin/Development
```
